app/models/concerns/smart_proxy_reference.rb

```diff
@@ -0,0 +1,51 @@
+module SmartProxyReference
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    attr_accessor :smart_proxy_references
+
+    def smart_proxy_reference(hash)
+      raise Foreman::Exception.new(N_("Proxy reference should be { :relation => [:column_name, ...]}")) unless correct_reference_format?(hash)
+      @smart_proxy_references ||= {}
```

Comment:
IMHO The code would be a bit more readable if we use `OpenStruct` instead of plain hash here...
I won't fight over it though...